### DISCUSSION TITLE
Superset and Hive connectivity with http thrift transport

### DISCUSSION THREAD

**Discussion by radhakvnr** [apache/superset#14042] [Q&A / Help]
Guys,

Badly need your help here.. We are using Superset 1.0.1 version and trying to connect to hive over http Thrift transport with authentication as LDAP. But ending with  below error. we are using Docker compose and pyhive is already there. For normal hive server (binary mode) with authentication =NOSASL we were able to connect without any issues where as for HTTP hive we were not able to connect.

The SQLAlchemy URI is hive://Username:Password@hostname:8443/default?auth=LDAP

Database.get_sqla_engine(). Masked URL: hive://username:XXXXXXXXXX@hostname:8443/default?auth=LDAP
[stats_logger] (incr) DatabaseRestApi.test_connection.error
[stats_logger] (timing) DatabaseRestApi.test_connection.time | 9.271875023841858
172.20.224.2 - - [21/Mar/2021 18:18:50] "POST /api/v1/database/test_connection HTTP/1.1" 422 -

Any help is highly appreciate here...

**Comment by radhakvnr** [comment]
A quick update from superset 0.38 version we were able to connect to hive without issues where as in 1.0.1 we are having issues

**Comment by clp007** [comment]
I encountered the same problem. Did you solve it? i have no auth
uri: hive://hostname:10000/default?auth=NONE

**Comment by radhakvnr** [comment]
Hi, if there is no authentication then try to use NOSASL it should work
If authentication is there then 
hive://username:password@host:port/default?auth=CUSTOM. Before doing this you need to follow.
https://github.com/apache/superset/pull/13608
